macros/src/main/scala/com/softwaremill/macwire/dependencyLookup/ValuesOfTypeInEnclosingClassFinder.scala

```diff
@@ -27,64 +31,187 @@ private[dependencyLookup] class ValuesOfTypeInEnclosingClassFinder[C <: Context]
+      val matches = debug.withBlock(s"Looking up imports in [$tree]") {
 
-              doFind(tail, treeToAdd :: acc)
-            } else doFind(tail, acc)
-
-          case Import(expr, selectors) =>
-            val matches = debug.withBlock(s"Looking up imports in [$tree]") {
-
-              val importCandidates : Map[Symbol, Name] =
-                if (selectors.exists { selector => selector.name.toString == "_" }) {
-                  // wildcard import on `expr`
-                  typeCheckUtil.typeCheckIfNeeded(expr).members.filter { _.isPublic }.map {
-                    s => s -> s.name.decodedName }.toMap
-                } else {
-                  val selectorNames = selectors.map(s => s.name -> s.rename).toMap
-                  typeCheckUtil.typeCheckIfNeeded(expr).
-                    members.
-                    collect { case m if selectorNames.contains(m.name) =>
-                      m -> selectorNames(m.name) }.
-                    toMap
-                }
-
-              filterImportMembers(importCandidates)
+        val importCandidates: List[(Symbol, (Tree, Tree))] =
+          (if (selectors.exists { selector => selector.name.toString == "_" }) {
+            // wildcard import on `expr`
+            typeCheckIfNeeded(expr).members.filter {
+              _.isPublic
+            }.map {
+              s => s -> s.name.decodedName
+            }
+          } else {
+            val selectorNames = selectors.map(s => s.name -> s.rename).toMap
+            typeCheckIfNeeded(expr).
+              members.
+              collect { case m if selectorNames.contains(m.name) =>
+              m -> selectorNames(m.name)
             }
-            doFind(tail, matches ::: acc)
+          }).map { case (s, name) => (s, (q"$expr.$s", Ident(name))) }.toList
+
+
+        filterImportMembers(importCandidates)
+      }
+      matches
+    }
 
-          case _ => doFind(tail, acc)
+    def evaluateValOrDef(tree: c.universe.Tree, name: c.universe.TermName, tpt: c.universe.Tree, rhs: c.universe.Tree, symbol: c.universe.Symbol): List[(c.universe.Tree, c.universe.Tree)] = {
+      val candidateOk = checkCandidate(t, name, tpt, treeToCheck(tree, rhs),
+        if (symbol.isMethod) "def" else "val")
+
+      if (candidateOk) {
+        val treeToAdd = implicitValue match {
+          case Some(i) if positionUtil.samePosition(i.symbol.pos, symbol.pos) => i
+          case _ => Ident(name)
+        }
+
+        List(treeToAdd -> treeToAdd)
+      } else {
+
+        // it might be a @Module, let's see
+        val hasSymbol = tpt.symbol != null // sometimes tpt has no symbol...
+        val valIsModule = hasSymbol && hasModuleAnnotation(tpt.symbol)
+        // the java @Inherited meta-annotation does not seem to be understood by scala-reflect...
+        val valParentIsModule = hasSymbol && !valIsModule && typeCheckIfNeeded(tpt).baseClasses.exists(hasModuleAnnotation)
```

Comment:
So here the compile-time retention would be useful, right? Not even for libraries, but for multi-module projects